models/ConfigureForm.php

```diff
@@ -207,6 +221,10 @@ public function saveSettings()
 
         $module->settings->set('enableEmailGoService', $this->enableEmailGoService);
         $module->settings->set('senderId', $this->senderId);
+        $module->settings->set('firebaseProjectId', $this->firebaseProjectId);
```

Comment:
@yurabakhtin Are you sure we need this information? The JSON file already contains the project ID and keys, for example, doesn't it?
e.g.

```json
{
  "type": "service_account",
  "project_id": "humhub-community",
  "private_key_id": "7acacacacacacacacacacacacacacacacacacacacacacacacacac",
  "private_key": "-----BEGIN PRIVATE KEY-----\nMIIXXXXXXXXXXXXXXXXXXXXXXXXXx/q\n-----END PRIVATE KEY-----\n",
  "client_email": "firebase-adminsdk-6ynpq@humhub-community.iam.gserviceaccount.com",
  "client_id": "12343262336346346346346346346",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/firebase-adminsdk-6ynpq%40humhub-community.iam.gserviceaccount.com",
  "universe_domain": "googleapis.com"
}
```

Reply:
@luke- If I try to initialize the app without `projectId` then I get the error:

`Could not initialize module: humhub.modules.firebase FirebaseError: Installations: Missing App configuration value: "projectId" (installations/missing-app-config-values).`

or do you mean we should extract the `project_id` from the provided JSON file?

Reply:
Yes, maybe I would be good to extract it.

Reply:
@luke- Commit https://github.com/humhub/fcm-push/pull/51/commits/56b3f926fdb71f8902600c26740eed1d641a63a6